File: Tarea #1 Lenguajes de Programacion 05/09/2024/Ejercicio2.cpp
#include <iostream>
#include <cstdlib>  
#include <ctime>    
#include <chrono>   

using namespace std;

int main() {
    int pares = 0, impares = 0;

    auto inicio = chrono::high_resolution_clock::now();

    srand(time(0));

    for (int i = 0; i < 500; i++) {
        int num = 50 + rand() % 51;  

        if (num % 2 == 0) {
            pares++;
        } else {
            impares++;
        }
    }

    cout << "Valores pares: " << pares << endl;
    cout << "Valores impares: " << impares << endl;

    auto fin = chrono::high_resolution_clock::now();
    chrono::duration<double> duracion = fin - inicio;

    cout << "Tiempo de ejecución: " << duracion.count() << " segundos" << endl;

    return 0;
}

File: Tarea #1 Lenguajes de Programacion 05/09/2024/Ejercicio2.py
import random
import time

start_time = time.perf_counter()

valores = [random.randint(50, 100) for _ in range(500)]

pares = len([num for num in valores if num % 2 == 0])
impares = len([num for num in valores if num % 2 != 0])

execution_time = time.perf_counter() - start_time

print(f"Pares: {pares}")
print(f"Impares: {impares}")
print(f"Tiempo de ejecución: {execution_time:.10f} segundos")